File: CitizenScienceBackend/src/main/java/com/citizenScience/dto/UserResponse.java
package com.citizenScience.dto;

import com.citizenScience.entities.User;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

/**
 * Data Transfer Object for user information responses.
 * Contains user details without sensitive information.
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserResponse {
    private UUID id;
    private String nome;
    private String cognome;
    private String email;
    private String ruolo;

    /**
     * Creates a UserResponse from a User entity.
     * 
     * @param user the user entity
     * @return UserResponse instance
     */
    public static UserResponse from(User user) {
        return UserResponse.builder()
                .id(user.getId())
                .nome(user.getNome())
                .cognome(user.getCognome())
                .email(user.getEmail())
                .ruolo(user.getRuolo())
                .build();
    }
}

File: citizen_science/lib/dto/user_response.dart
/// Data transfer object for user responses from the API.
///
/// Contains user profile information returned from authentication
/// and user management endpoints.
class UserResponse {
  final String id;
  final String nome;
  final String cognome;
  final String email;
  final String ruolo;

  UserResponse({
    required this.id,
    required this.nome,
    required this.cognome,
    required this.email,
    required this.ruolo,
  });

  /// Creates a [UserResponse] from a JSON map.
  factory UserResponse.fromJson(Map<String, dynamic> json) {
    return UserResponse(
      id: json['id'],
      nome: json['nome'],
      cognome: json['cognome'],
      email: json['email'],
      ruolo: json['ruolo'],
    );
  }

  /// Converts this [UserResponse] to a JSON map.
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nome': nome,
      'cognome': cognome,
      'email': email,
      'ruolo': ruolo,
    };
  }

  /// Returns the full name combining nome and cognome.
  String get fullName => '$nome $cognome';
}
